provider/docker/docker.go

```diff
@@ -155,33 +155,49 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 				Configuration: configuration,
 			}
 			if p.Watch {
-				ctx, cancel := context.WithCancel(ctx)
 				if p.SwarmMode {
+					eventsCtx := context.Background()
+					eventsMsgChan, eventsErrorChan := dockerClient.Events(
+						eventsCtx,
+						dockertypes.EventsOptions{
+							Filters: filters.NewArgs(filters.Arg("scope", "swarm")),
+						},
```

Comment:
`swarm` events can have many types (`service`, `node`,`secret`).
WDYT about adding a filter on the `type `service`?

Thus Træfik will only updates its list when the event is about an action on a service.

Reply:
Good point. 😄